src/probnum/diffeq/_perturbsolve_ivp.py

```diff
@@ -0,0 +1,191 @@
+        Optional. Default is ``1e-4``.
+    step :
+        Step size. If atol and rtol are not specified, this step-size is used for a fixed-step ODE solver.
+        If they are specified, this only affects the first step. Optional.
+        Default is None, in which case the first step is chosen as prescribed by :meth:`propose_firststep`.
+
+    Examples
+    --------
+    >>> from probnum.diffeq import perturbsolve_ivp
+    >>> import numpy as np
+
+    Solve a simple logistic ODE with fixed steps.
+
+    >>> rng = np.random.default_rng(seed=2)
+    >>>
+    >>> def f(t, x):
+    ...     return 4*x*(1-x)
+    >>>
+    >>> y0 = np.array([0.15])
+    >>> t0, tmax = 0., 1.5
+    >>> solution = perturbsolve_ivp(f, t0, tmax, y0, rng=rng, step=0.25, method="RK23", adaptive=False)
+    >>> print(np.round(solution.states.mean, 3))
+    [[0.15 ]
+     [0.292]
+     [0.497]
+     [0.766]
+     [0.874]
+     [0.955]
+     [0.987]]
+
+    Each solution is the result of a randomly-perturbed call the an underlying Runge-Kutta solver.
+    Therefore, if you call it again, the output will be different:
+
+    >>> other_solution = perturbsolve_ivp(f, t0, tmax, y0, rng=rng, step=0.25, method="RK23", adaptive=False)
+    >>> print(np.round(other_solution.states.mean, 3))
+    [[0.15 ]
+     [0.282]
+     [0.448]
+     [0.657]
+     [0.853]
+     [0.945]
+     [0.975]]
+
+    Other methods, such as `RK45` (as well as other perturbation styles) are easily accessible.
+    Let us solve the same equation, with an adaptive RK45 solver that uses lognormally perturbed steps.
+
+    >>> solution = perturbsolve_ivp(f, t0, tmax, y0, rng=rng, atol=1e-5, rtol=1e-6, method="RK45", adaptive=True)
+    >>> print(np.round(solution.states.mean, 3))
+    [[0.15 ]
+     [0.152]
+     [0.167]
+     [0.26 ]
+     [0.431]
+     [0.646]
+     [0.849]
+     [0.883]
+     [0.915]
+     [0.953]
+     [0.976]
+     [0.986]]
+
+    References
+    ----------
+    .. [1] Abdulle, A. and Garegnani, G..
+        Random time step probabilistic methods for uncertainty quantification in chaotic and geometric numerical integration.
+        Statistics and Computing. 2020.
+    .. [2] J. R. Dormand, P. J. Prince..
+        A family of embedded Runge-Kutta formulae.
+        Journal of Computational and Applied Mathematics, Vol. 6, No. 1, pp. 19-26, 1980.
+    .. [3] L. W. Shampine.
+        Some Practical Runge-Kutta Formulas.
+        Mathematics of Computation, Vol. 46, No. 173, pp. 135-150, 1986.
+    .. [4] P. Bogacki, L.F. Shampine.
+        A 3(2) Pair of Runge-Kutta Formulas.
+        Appl. Math. Lett. Vol. 2, No. 4. pp. 321-325, 1989.
+    """
+
+    if method not in METHODS:
+        raise ValueError("Method is not supported.")
+
+    if perturb not in PERTURBS:
+        raise ValueError("Perturbation-style is not supported.")
+
+    ivp = problems.InitialValueProblem(t0=t0, tmax=tmax, y0=np.asarray(y0), f=f)
+    scipy_solver = SELECT_METHOD[method](ivp.f, ivp.t0, ivp.y0, ivp.tmax)
```

Comment:
Is there a benefit to constructing the `ivp`, just to deconstruct it again? Maybe that line could be removed.

Reply:
This is because the scipy solver need (for some reason) the whole IVP at construction time. 
In #486 this will be resolved entirely.

Reply:
I don't see where the scipy solver has access to the ivp object? But I found another answer to my question: In line 186, `stepsize.propose_firststep` requires access to an `ivp` object, so that's a reason for building it.

Reply:
Ah, I misunderstood your question then... Glad it is resolved anyway :)